### PR TITLE
Add new episode on GitHub Actions and Docker

### PR DESCRIPTION
**Pull request by jhidding**
Rationale for this PR was set out in issue #76.

We will teach this content on Tuesday 23 March, after which we may have some updates to this PR. This should also give us more insight into the feasibility of teaching this material in the scope of an introduction to Docker (over which some concerns were raised in #76)

### CODE REVIEW

**Review by sstevens2 — CHANGES_REQUESTED** [review]
Overall I would suggest adding more text on why you want to make this website and why you want to setup github action to do it.  The story at present is kind of missing from the ~lesson~episode and it is hard to see how this is an authentic task that someone might want to do.
I couldn't give much feedback on the GitHub actions section as I don't know it well enough to do so.  I am concerned the learners will probably get caught up in what GitHub is, what GitHub actions are, why you would want to make a webpage like this, and the syntax of what they are entering in the GitHub actions part.

For this to be included in extras you will need to move it from `_episodes` to the `_extras` folder.

File: _episodes/e01-github-actions.md
Comment (jhidding) [reply _episodes/e01-github-actions.md:44]: All of these are covered in Software Carpentry. I will add some text and a link to http://swcarpentry.github.io/git-novice/07-github/index.html.

File: _episodes/e01-github-actions.md
Comment (jhidding) [reply _episodes/e01-github-actions.md:117]: Yes, I thought about that. The thing is that `pandoc` command lines tend to grow out of hand anyway. In practice these commands would be part of some Makefile or other. I decided on the `--metadata` line for exactly the reason you mentioned.

**Comment by jhidding** [comment]
@sstevens2 Thanks for the review! I incorporated your suggestions.

**Comment by jhidding** [comment]
One remark on the `_extras` folder: in the current template `_extras` is not a very visible place to put extra lesson content. I'm afraid no one will even spot it if its there. By putting "(extra)" in front of the title, at least it is clear to everyone that this episode is not part of the core lesson.

File: _episodes/e01-github-actions.md
Comment (dmwelch) [reply _episodes/e01-github-actions.md]: Adding the Git novice lesson as a prerequisite is a definite requirement.

Comment (sstevens2) [reply _episodes/e01-github-actions.md]: Looks like this is mentioned as a callout at the top of the lesson.

**Comment by sstevens2** [comment]
@jhidding Any feedback or changes to this episode based on your recent pilot?

Also a new issue or email about how the pilot went would be very welcome.

**Comment by jhidding** [comment]
Sure! We had one or two persons for who it wasn't entirely clear why this was useful, it is unclear if that's a presentation or content issue. Most people really enjoyed it as a concrete example of what you can do with Docker containers and a few responded that the episode gave them ideas for trying out other things with Github Actions + Docker.

Content wise, the episode introduced quite a bit of concepts but even if people didn't get everything, they could still enjoy it as a demo of a cool application.

**Comment by dme26** [comment]
Agreed @jhidding that `_extras` potentially risks content being seen in some lessons, but within this lesson we're progressively moving more content over to there, and hopefully adding further such episodes, given the many Docker paths that learners may wish to take. If you are happy to update your PR to place the content in `_extras` as @sstevens2 has recommended, it would be great to have your content included.

**Comment by jhidding** [comment]
> Agreed @jhidding that `_extras` potentially risks content being seen in some lessons, but within this lesson we're progressively moving more content over to there, and hopefully adding further such episodes, given the many Docker paths that learners may wish to take. If you are happy to update your PR to place the content in `_extras` as @sstevens2 has recommended, it would be great to have your content included.

One more problem I spotted, is that the content in `_extras` is rendered from a different template: questions, objectives and keypoints are not included in the rendered HTML. Fact remains that in the current carpentries template, lesson material should not be put in `_extras`. I will comply with your suggestion, and leave it up to maintainers to decide what to do with it.

**Comment by tobyhodges** [comment]
> the content in _extras is rendered from a different template: questions, objectives and keypoints are not included in the rendered HTML.

This is indeed the case by default, but can be over-ridden by specifying the `episode` layout for the Markdown file in `_extras` i.e.

```yaml
---
title: "Using Docker with Github Actions"
layout: episode
teaching: 30
exercises: 0
questions:
- "How do I use Docker from Github Actions?"
objectives:
- "Generate Github.io pages using Pandoc"
- "Raise awareness of how Docker can be used in cloud services"
keypoints:
- "You can call any Docker image from a Github action"
---
```

**Review by sstevens2 — APPROVED** [review]

**Comment by sstevens2** [comment]
Thanks for this addition @jhidding!  In our upcoming sprint in June we are going to rework the "Containers used in this lesson" to point to the example use cases in extras and this will be good to include.

**Pull request merged**
